src/sss_client/idmap/sss_nss_idmap.c

```diff
@@ -324,6 +324,11 @@ static int sss_nss_getyyybyxxx(union input inp, enum sss_cli_command cmd,
         goto done;
     }
 
+    if (replen < DATA_START) { /* make sure 'type' is present */
```

Comment:
This check could just replace the check in line 313. This one is more restrictive that the existing one.

Reply:
It actually can't, at least not straightforward.

'replen == 8'  => basic header provided but no results => `return ENOENT` (not EBADMSG)

I agree resulting code is convoluted but...

Reply:
You are right. Missed that case.